### PR TITLE
Sockets - Nidhi 

### PR DESCRIPTION
**Pull request by nidhiparixitpatel**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | In Rails, the Model represents the data that we are working with and manages the structure and relationships of this data.
Describe in your own words what the Controller is doing in Rails | In Rails, the Controller acts as the manager of the requests that get sent in and manipulating data and directing what is sent back to the request.
Describe in your own words what the View is doing in Rails | In Rails, the View holds what is displayed and what a user can interact with. 
Describe an edge-case controller test you wrote | For marking complete and incomplete, I checked that the page redirects to the tasklist page if given an invalid task to mark. 
What is the purpose of using strong params? (i.e. the params method in the controller) | Strong params in the controller help validate what parameters are sent in and DRY up code. 
How are Rails migrations related to Rails models? | Rails migrations change the structure of a table which Rails Models use to fill in information.
Describe one area of Rails that are still unclear on | I am still unsure about routes and how to decide naming and parameters associated with paths

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  def destroy
+    task_id = params[:id]
+    @task = Task.find_by(id: task_id)
+  
+    if @task
+      @task.destroy
+      redirect_to tasks_path
+    else
+      render :notfound, status: :not_found
```
Comment (CheezItMan) [inline app/controllers/tasks_controller.rb:72]: 👍

File: test/controllers/tasks_controller_test.rb [hunk 2]
```diff
@@ -105,24 +110,110 @@
...
+
+    end
   end
 
   # Complete these tests for Wave 4
   describe "destroy" do
-    # Your tests go here
+    
+    
+    it "can destroy a task" do
```
Comment (CheezItMan) [inline test/controllers/tasks_controller_test.rb:158]: You should also test what happens when you try to delete a task that doesn't exist.

**Comment by CheezItMan** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | You should be committing more, but otherwise good
Answered comprehension questions | Check, hopefully last week's practice helped with naming parameters in paths.  
Successfully handles: Index, Show | Check
Index & Show tests pass | Check
Successfully handles: New, Create | Check
New & Create tests pass | Check
Successfully handles: Edit, Update | Check
Tests for Edit & Update test valid & invalid task ids | Check
Successfully handles: Destroy, Task Complete | Check
Tests for Destroy & Task Complete include tests for valid and invalid task ids | Check
Routes follow RESTful conventions | Check
Uses named routes (like `_path`) | Check
**Overall** | Nice work, it's simple and unstyled, but functions and does everything required.
